tests/functional/steps/then.py

```diff
@@ -114,3 +115,9 @@ def check_sensitive_fields_are_absent(response_body: dict) -> None:
     with check:
         for field in _sensitive_fields:
             assert not is_key_in_dict(response_body, field), f'Sensitive field, {field}, in response.'
+
+
+@then("the response body contains the record's new version number")
+def version_incremented(response_body: Response, update: Update) -> None:
+    with check:
+        assert response_body["meta"]["versionId"] == str(int(update.record_version) + 1)
```

Comment:
This was moved from test_user_restricted.py so that it could be used as a common fixture across other test modules.